File: app/src/main/java/com/yawar/memo/modelView/IntroActModelView.java
package com.yawar.memo.modelView;

import android.annotation.SuppressLint;
import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import java.util.ArrayList;


import android.app.Application;
import android.content.Context;

import androidx.annotation.NonNull;
        import androidx.lifecycle.AndroidViewModel;
        import androidx.lifecycle.LiveData;
        import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.yawar.memo.constant.AllConstants;
import com.yawar.memo.model.ChatRoomModel;
import com.yawar.memo.model.UserModel;
import com.yawar.memo.repositry.BlockUserRepo;
import com.yawar.memo.repositry.ChatRoomRepo;
import com.yawar.memo.retrofit.RetrofitClient;
import com.yawar.memo.utils.BaseApp;

import org.json.JSONObject;

import java.util.ArrayList;
        import java.util.List;

import io.reactivex.Single;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

public class IntroActModelView extends ViewModel {

    BaseApp baseApp = BaseApp.getInstance();
    private final ChatRoomRepo repository = baseApp.getChatRoomRepo();
    private final BlockUserRepo blockUserRepo = baseApp.getBlockUserRepo();



    public  MutableLiveData<ArrayList<ChatRoomModel>> chatRoomListMutableLiveData= new MutableLiveData<>();
    public final ArrayList<ChatRoomModel> chatRoomsList = new ArrayList<>();
    public final MutableLiveData<ArrayList<UserModel>> userBlockListMutableLiveData= new MutableLiveData<>();


    public IntroActModelView() {
        repository.callAPI(baseApp.getClassSharedPreferences().getUser().getUserId());


    }





    public  MutableLiveData<ArrayList<ChatRoomModel>> loadData() {
//        chatRoomListMutableLiveData.setValue((ArrayList<ChatRoomModel>) repository.getChatRoomModelList());
        return repository.getChatRoomListMutableLiveData();
//        System.out.println(chatRoomListMutableLiveData.getValue().size()+"chatRoomListMutableLiveData.getValue().size()");

    }
    public MutableLiveData<ArrayList<UserModel>> getUserBlock (){

//        userBlockListMutableLiveData.setValue( blockUserRepo.getUserBlockList());
        return blockUserRepo.getUserBlockList();


    }
    @SuppressLint("CheckResult")
    public void sendFcmToken(String user_id, String token){
        Single<String> observable = RetrofitClient.getInstance(AllConstants.base_node_url).getapi().sendFcmToken(user_id,token)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
        observable.subscribe(s -> {


        },s-> {
            System.out.println("Errorrrrrrrr" + s);
          });
    }

    public MutableLiveData<Boolean> getLoading(){
        return repository.loading;
    }

    public MutableLiveData<Boolean> getErrorMessage(){
        return repository.showErrorMessage;
    }
    public void setLoading(Boolean check){
        repository.loading.setValue(check);
    }
    public void setErrorMessage(Boolean check){
        repository.showErrorMessage.setValue(check);
    }

    @Override
    protected void onCleared() {
        super.onCleared();
    }
}

File: app/src/main/java/com/yawar/memo/ui/introPage/IntroActModelView.kt
package com.yawar.memo.ui.introPage

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.yawar.memo.BaseApp
import com.yawar.memo.domain.model.ChatRoomModel
import com.yawar.memo.repositry.ChatRoomRepoo
import dagger.hilt.android.lifecycle.HiltViewModel
import java.util.*
import javax.inject.Inject
@HiltViewModel
 class IntroActModelView @Inject  constructor( val chatRoomRepoo: ChatRoomRepoo) : ViewModel() {
//class IntroActModelView : ViewModel() {

    var baseApp = BaseApp.instance!!
//    private val repository = baseApp.chatRoomRepo
//    var chatRoomRepoo = baseApp.chatRoomRepoo
    var chatRoomListMutableLiveData = MutableLiveData<ArrayList<ChatRoomModel>>()
    init {
        Log.d("IntroActModelView", chatRoomRepoo.chatRoomListMutableLiveData.toString())
            if(chatRoomRepoo.chatRoomListMutableLiveData.value == null  ) {
                Log.d("IntroActModelView2", chatRoomRepoo.chatRoomListMutableLiveData.toString())
                    chatRoomRepoo.loadChatRoom(baseApp.classSharedPreferences?.user?.userId!!)

            }
        else if(chatRoomRepoo.chatRoomListMutableLiveData.value!!.isEmpty()){
                chatRoomRepoo.loadChatRoom(baseApp.classSharedPreferences?.user?.userId!!)

            }
    }
    fun loadData(): LiveData<ArrayList<ChatRoomModel?>?> {
        return chatRoomRepoo.chatRoomListMutableLiveData as  LiveData<ArrayList<ChatRoomModel?>?>
    }


    fun setLoading(check: Boolean) {
        chatRoomRepoo.setLoading(check)
    }

    fun setErrorMessage(check: Boolean) {
        chatRoomRepoo.setErrorMessage(check)
    }

    fun getLoading(): LiveData<Boolean> {
        return chatRoomRepoo.loadingMutableLiveData;
    }

    fun getErrorMessage(): LiveData<Boolean> {
        return chatRoomRepoo.showErrorMessage;
    }


}